rotkehlchen/api/rest.py

```diff
@@ -4416,3 +4417,49 @@ def search_for_names_everywhere(self, addresses: List[ChecksumEthAddress]) -> Re
             addresses=addresses,
         )
         return api_response(_wrap_in_ok_result(mappings))
+
+    def _detect_ethereum_tokens(
+            self,
+            only_cache: bool,
+            addresses: List[ChecksumEthAddress],
+    ) -> Dict[str, Any]:
+        ethtokens = EthTokens(
+            database=self.rotkehlchen.data.db,
+            ethereum=self.rotkehlchen.chain_manager.ethereum,
+        )
+        try:
+            with self.rotkehlchen.data.db.user_write() as cursor:
+                account_tokens_info = ethtokens.detect_tokens(
+                    write_cursor=cursor,
+                    only_cache=only_cache,
+                    accounts=addresses,
+                )
+        except (RemoteError, BadFunctionCallOutput) as e:
+            return wrap_in_fail_result(message=str(e), status_code=HTTPStatus.CONFLICT)
+        result = {}
+        for account, (tokens, last_update_ts) in account_tokens_info.items():
+            result[account] = {
+                'tokens': [token.identifier for token in tokens] if tokens is not None else None,
+                'last_update_timestamp': last_update_ts,
+            }
+        return _wrap_in_ok_result(result)
```

Comment:
_wrap_in_ok_result does not put a status code! You need a status code here, so either do the wrapping manually or add a default argument to `_wrap_in_ok_result` just like `wrap_in_fail_result` has

Reply:
`_get_status_code_from_async_response` which is used further down the line has the `status_code` parameter with default value `OK`.

But no problem, I will add the status code here too to be more explicit.